src/masternodeman.cpp

```diff
@@ -858,6 +869,17 @@ int CMasternodeMan::ProcessMessageInner(CNode* pfrom, std::string& strCommand, C
         vRecv >> mnb;
         return ProcessMNBroadcast(pfrom, mnb);
 
+    } else if (strCommand == NetMsgType::MNBROADCAST2) {
+        if (!Params().GetConsensus().NetworkUpgradeActive(GetBestHeight(), Consensus::UPGRADE_V5_3)) {
+            LogPrint(BCLog::MASTERNODE, "%s: mnb2 not enabled pre-V5.3 enforcement\n", __func__);
+            return 30;
+        }
+        CMasternodeBroadcast mnb;
+        OverrideStream<CDataStream> s(&vRecv, vRecv.GetType(), vRecv.GetVersion() | ADDRV2_FORMAT);
+        s >> mnb;
+        mnb.isBIP155Addr = !mnb.addr.IsAddrV1Compatible();
+        return ProcessMNBroadcast(pfrom, mnb);
```

Comment:
We could even reject the `MNBROADCAST2` if `mnb.addr.IsAddrV1Compatible()`, and process it only if `mnb.isBIP155Addr == true`.

Reply:
Sure, we can do that.